Windows joypad code is using DWORDs (unsigned int) instead of LONGs for device constants

<!-- Please search existing issues for potential duplicates before filing yours:
https://github.com/godotengine/godot/issues?q=is%3Aissue
-->

**Godot version:**
master 494bf38c76d3070c548a104d56d9f7e164226f1c
3.2 d3ffff3ad8552b4f75c96f4ff43451b8f60fec5e

**OS/device including version:**
Windows 10 2004 (May 2020)

**Issue description:**
Compiling on Windows with clang 11.0.0, throws the following error:
```
platform\\windows\\joypad_windows.cpp:398:27: error: non-constant-expression cannot be narrowed from type 'LONG' (aka 'long') to 'unsigned int' in initializer list [-Wc++11-narrowing]
unsigned int axes[] = { DIJOFS_X, DIJOFS_Y, DIJOFS_Z, DIJOFS_RX, DIJOFS_RY, DIJOFS_RZ, DIJOFS_SLIDER(0), DIJOFS_SLIDER(1) };
```

**Steps to reproduce:**
Try compile Godot on Windows with clang.

**Minimal reproduction project:**
N/A

Note: The `DIJOFS*` constants are actually macros that extract the values from the `DIJOYSTATE` `struct` where they are [defined ](https://docs.microsoft.com/en-us/previous-versions/windows/desktop/ee416627(v=vs.85)) as LONGs. However, Godot is incorrectly using `DWORD`s or `unsigned int`s when referencing these values.

--- a/platform/windows/joypad_windows.cpp
+++ b/platform/windows/joypad_windows.cpp
@@ -194,7 +194,7 @@ void JoypadWindows::setup_joypad_object(const DIDEVICEOBJECTINSTANCE *ob, int p_
 		HRESULT res;
 		DIPROPRANGE prop_range;
 		DIPROPDWORD dilong;
-		DWORD ofs;
+		LONG ofs;
 		if (ob->guidType == GUID_XAxis)
 			ofs = DIJOFS_X;
 		else if (ob->guidType == GUID_YAxis)
@@ -395,7 +395,7 @@ void JoypadWindows::process_joypads() {
 
 		// on mingw, these constants are not constants
 		int count = 8;
-		unsigned int axes[] = { DIJOFS_X, DIJOFS_Y, DIJOFS_Z, DIJOFS_RX, DIJOFS_RY, DIJOFS_RZ, DIJOFS_SLIDER(0), DIJOFS_SLIDER(1) };
+		LONG axes[] = { DIJOFS_X, DIJOFS_Y, DIJOFS_Z, DIJOFS_RX, DIJOFS_RY, DIJOFS_RZ, (LONG)DIJOFS_SLIDER(0), (LONG)DIJOFS_SLIDER(1) };
 		int values[] = { js.lX, js.lY, js.lZ, js.lRx, js.lRy, js.lRz, js.rglSlider[0], js.rglSlider[1] };
 
 		for (int j = 0; j < joy->joy_axis.size(); j++) {

--- a/platform/windows/joypad_windows.h
+++ b/platform/windows/joypad_windows.h
@@ -77,7 +77,7 @@ private:
 		DWORD last_pad;
 
 		LPDIRECTINPUTDEVICE8 di_joy;
-		List<DWORD> joy_axis;
+		List<LONG> joy_axis;
 		GUID guid;
 
 		dinput_gamepad() {

